packages/office-ui-fabric-react/src/components/pickers/BasePicker.tsx

```diff
@@ -194,10 +203,44 @@ export class BasePicker<T, P extends IBasePickerProps<T>> extends BaseComponent<
   }
 
   protected updateSuggestions(suggestions: any[]) {
-    this.suggestionStore.updateSuggestions(suggestions);
+    this.suggestionStore.updateSuggestions(suggestions, true);
     this.forceUpdate();
   }
 
+  protected onEmptyInputFocus() {
+    let suggestions: T[] | PromiseLike<T[]> = this.props.onInputFocus(this.state.items);
+    let suggestionsArray: T[] = suggestions as T[];
+    let suggestionsPromiseLike: PromiseLike<T[]> = suggestions as PromiseLike<T[]>;
+    if (Array.isArray(suggestionsArray)) {
+      this.suggestionStore.updateSuggestions(suggestionsArray, false);
+    } else if (suggestionsPromiseLike && suggestionsPromiseLike.then) {
+      this.setState({
+        suggestionsLoading: true
+      });
+
+      // Clear suggestions
+      this.suggestionStore.updateSuggestions([], false);
+
+      this.setState({
+        suggestionsVisible: this.input.inputElement === document.activeElement
+      });
+      // Ensure that the promise will only use the callback if it was the most recent one.
+      let promise: PromiseLike<T[]> = this.currentPromise = suggestionsPromiseLike;
+      promise.then((newSuggestions: T[]) => {
+        if (promise === this.currentPromise) {
+          this.suggestionStore.updateSuggestions(newSuggestions, false);
+          this.setState({
+            suggestionsLoading: false
+          });
+          if (this.loadingTimer) {
+            this._async.clearTimeout(this.loadingTimer);
+            this.loadingTimer = undefined;
+          }
+        }
+      });
```

Comment:
Most of this is duplicated below in updateValue. It should be factored out into a separate function.